src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -287,40 +287,41 @@ static smt_termt convert_relational_to_smt(
     binary_relation.pretty());
 }
 
-static smt_termt
-convert_expr_to_smt(const binary_relation_exprt &binary_relation)
+static optionalt<smt_termt> try_relational_conversion(const exprt &expr)
```

Comment:
Since this is effectively a sub-dispatcher to the more general `convert_expr_to_smt`, wouldn't it make more sense to flatten it in there (by removing this method and moving its sub-dispatch to `convert_expr_to_smt`)?

Or is there a specific reason for this to stand as a functional unit on its own?

Reply:
Because the old `smt2_convt` implementation separates these into a separate `convert_relation` function. Keeping a similar structure facilitates cross checking between the two implementations.